libs/core/src/lib/button/button-host.directive.ts

```diff
@@ -0,0 +1,183 @@
+    | 'emphasized'
+    | 'menu';
+
+@Directive({
+    // eslint-disable-next-line @angular-eslint/directive-selector
+    selector: '[fd-base-button]',
+    host: {
+        '[class]': '_computedClassName()',
+        '[attr.type]': 'type()',
+        '[attr.disabled]': 'disabled() || null',
+        '[attr.aria-description]': 'buttonAriaDescription()',
+        '[attr.aria-pressed]': 'toggled()',
+        '[style.pointer-events]': 'disabled() ? "none" : ""'
+    },
+    providers: [contentDensityObserverProviders()],
+    standalone: true
+})
+export class ButtonHostDirective {
+    /** The property allows user to pass additional css classes. */
+    @Input('class')
+    set _class(value: string) {
+        this.class.set(value);
+    }
+
+    /**
+     * Native type of button element
+     */
+    @Input('type')
+    set _type(value: Nullable<string>) {
+        this.type.set(value || 'button');
+    }
+
+    /** The type of the button. Types include:
+     * 'standard' | 'positive' | 'negative' | 'attention' | 'half' | 'ghost' | 'transparent' | 'emphasized' | 'menu'.
+     * Leave empty for default (Standard button).'
+     * Default value is set to 'standard'
+     */
+    @Input('fdType')
+    set _fdType(value: ButtonType) {
+        this.fdType.set(value || 'standard');
+    }
+
+    /**
+     * Text rendered inside button component
+     */
+    @Input('label')
+    set _label(value: string) {
+        this.label.set(value);
+    }
+
+    /** Whether to apply menu mode to the button.
+     * Default value is set to false
+     */
+    @Input('fdMenu')
+    set _fdMenu(value: Nullable<boolean>) {
+        this.fdMenu.set(value || false);
+    }
+
+    /** adding native aria-label to the component */
+    @Input('ariaLabel')
+    set _ariaLabel(value: Nullable<string>) {
+        this.ariaLabel.set(value);
+    }
+
+    /** adding native aria-description to the componenet */
+    @Input('ariaDescription')
+    set _ariaDescription(value: Nullable<string>) {
+        this.ariaDescription.set(value);
+    }
+
+    /** Whether button is in toggled state. */
+    @Input({ alias: 'toggled', transform: coerceBooleanProperty })
+    set _toggled(value: BooleanInput) {
+        this.toggled.set(value as boolean);
+    }
+
+    /**
+     * Native disabled attribute of button element
+     */
+    @Input({ alias: 'disabled', transform: coerceBooleanProperty })
+    set _disabled(value: BooleanInput) {
+        this.disabled.set(value as boolean);
+    }
+
+    /**
+     * Native aria-disabled attribute of button element
+     */
+    @Input({ alias: 'aria-disabled', transform: coerceBooleanProperty })
+    set _ariaDisabled(value: BooleanInput) {
+        this.ariaDisabled.set(value as boolean);
+    }
+
+    /** @hidden */
+    specialButtonType: Array<string> = ['emphasized', 'positive', 'negative', 'attention'];
+
+    /** @hidden */
+    elementRef = inject<ElementRef<HTMLElement>>(ElementRef);
+
+    /** @hidden */
+    class = signal('');
```

Comment:
Why not keep the input properties like `class`, `type`, `fdType` as is, without the `_` prefix, and the signals will have `_` prefix instead?